scripts/extractCodeSnippets.js

```diff
@@ -71,17 +72,26 @@ codeSnippets.forEach((snippets, operationId) => {
       ${joinSnippets(snippets)}
     })();
   `;
-  fs.writeFileSync(`${snippetsFolder}/${operationId}.ts`, codeToTest);
+  fs.writeFileSync(path.join(snippetsFolder, `${operationId}.ts`), codeToTest);
 });
 
-// write tsconfig file
+const tsconfig = {};
 tsconfig.include = ['*.ts'];
-tsconfig.compilerOptions.noUnusedLocals = false;
+tsconfig.compilerOptions = {
+  noUnusedLocals: false,
+  outDir: 'dist',
+  declaration: false,
+  sourceMap: false,
+};
 
-fs.writeFileSync(
-  snippetsFolder + '/tsconfig.json',
-  JSON.stringify(tsconfig, null, 2) + '\n'
-);
-console.log(
-  `TS config for code snippets saved to: ${snippetsFolder}/tsconfig.json`
-);
+// We make a tsconfig.build.json that extends the normal tsconfig.build.json
+tsconfig.extends = '../../../tsconfig.build.json';
+const tsbuildconfigpath = path.join(snippetsFolder, './tsconfig.build.json');
+fs.writeFileSync(tsbuildconfigpath, JSON.stringify(tsconfig, null, 2) + '\n');
+console.log(`TS build config for code snippets saved to: ${tsbuildconfigpath}`);
+
+// We also make a tsconfig.json for IDEs only that includes path aliases
```

Comment:
I also get this
```
Error: Cannot write file '/Users/polomanyi/Workspace/cognite-sdk-js/packages/core/package.json' because it would overwrite input file.
```

Reply:
I get that warning every time, don't know what it is about.
About the tsconfig, I make two of them, with the only difference being what tsconfig it inherits from, such that Go To Definition works from snippets as well

Reply:
> Go To Definition works from snippets as well

why do we need that at all?
code snippets need to be just tested and pushed to contracts

Reply:
> I get that warning every time, don't know what it is about

I guess you are literally trying to write to the file that exists

Reply:
I removed the `tsconfig.json` from the snippets folder, which was good because IDEs use the parent folder `tsconfig.json`. The snippets still have their own `tsconfig.build.json`